doozer/doozerlib/cli/scan_sources.py

```diff
@@ -3,126 +3,205 @@
-                        if not eldest_rpm_build or latest_rpm_build['creation_event_id'] < eldest_rpm_build['creation_event_id']:
+                    latest_rpms = koji_api.getLatestRPMS(tag=meta.branch() + '-candidate', package=package_name)[1]
+
+                    for latest_rpm_build in latest_rpms:
+                        if not eldest_rpm_build or latest_rpm_build['creation_event_id'] < \
+                                eldest_rpm_build['creation_event_id']:
                             eldest_rpm_build = latest_rpm_build
 
-                    # Detect if our buildroot changed since the oldest rpm of the latest build of the package was built.
-                    build_root_change = brew.has_tag_changed_since_build(runtime, koji_api, eldest_rpm_build, meta.build_root_tag(), inherit=True)
+                    # Detect if our buildroot changed since the oldest rpm of the package latest build of was built.
+                    build_root_change = brew.has_tag_changed_since_build(self.runtime, koji_api, eldest_rpm_build,
+                                                                         meta.build_root_tag(), inherit=True)
 
                     if build_root_change:
-                        rebuild_hint = RebuildHint(RebuildHintCode.BUILD_ROOT_CHANGING, 'Oldest package rpm build was before buildroot change')
-                        runtime.logger.info(f'{dgk} ({eldest_rpm_build}) in {package_name} is older than more recent buildroot change: {build_root_change}')
+                        rebuild_hint = RebuildHint(RebuildHintCode.BUILD_ROOT_CHANGING,
+                                                   'Oldest package rpm build was before buildroot change')
+                        self.runtime.logger.info(f'{dgk} ({eldest_rpm_build}) in {package_name} '
+                                                 f'is older than more recent buildroot change: {build_root_change}')
 
                 if rebuild_hint.rebuild:
-                    add_assessment_reason(meta, rebuild_hint)
-                    changing_rpm_metas.add(meta)
-                    changing_rpm_packages.add(package_name)
+                    self.add_assessment_reason(meta, rebuild_hint)
+                    self.changing_rpm_metas.add(meta)
+                    self.changing_rpm_packages.add(package_name)
 
             elif meta.meta_type == 'image':
                 if rebuild_hint.rebuild:
-                    add_image_meta_change(meta, rebuild_hint)
+                    self.add_image_meta_change(meta, rebuild_hint)
 
             else:
                 raise IOError(f'Unsupported meta type: {meta.meta_type}')
 
-        # To limit the size of the queries we are going to make, find the oldest and newest image
-        oldest_image_event_ts = None
-        newest_image_event_ts = 0
-        for image_meta in runtime.image_metas():
+    def find_oldest_newest(self, koji_api):
```

Comment:
This method does more than its name.

Reply:
I kept the same name just for the part that checks build date. Moved the rest to a better place, with more and shorter methods